plugins/parsers/nagios/parser_test.go

```diff
@@ -1,112 +1,377 @@
-
-	// Output3
-	metrics, err = parser.Parse([]byte(validOutput3))
-	require.NoError(t, err)
-	require.Len(t, metrics, 1)
-	// time
-	assert.Equal(t, "time", metrics[0].Tags()["perfdata"])
-	assert.Equal(t, map[string]interface{}{
-		"value": float64(0.008457),
-	}, metrics[0].Fields())
-	assert.Equal(t, map[string]string{"perfdata": "time"}, metrics[0].Tags())
-
-	// Output4
-	metrics, err = parser.Parse([]byte(validOutput4))
-	require.NoError(t, err)
-	require.Len(t, metrics, 3)
-	// load
-	// const validOutput4 = "OK: Load average: 0.00, 0.01, 0.05 | 'load1'=0.00;0:4;0:6;0; 'load5'=0.01;0:3;0:5;0; 'load15'=0.05;0:2;0:4;0;"
-	assert.Equal(t, map[string]interface{}{
-		"value":       float64(0.00),
-		"warning_lt":  MinFloat64,
-		"warning_gt":  float64(4),
-		"critical_le": float64(0),
-		"critical_ge": float64(6),
-		"min":         float64(0),
-	}, metrics[0].Fields())
-
-	assert.Equal(t, map[string]string{"perfdata": "load1"}, metrics[0].Tags())
+func assertNagiosState(t *testing.T, m telegraf.Metric, f map[string]interface{}) {
+	assert.Equal(t, map[string]string{}, m.Tags())
+	assert.Equal(t, f, m.Fields())
 }
 
-func TestParseInvalidOutput(t *testing.T) {
+func TestParse(t *testing.T) {
 	parser := NagiosParser{
 		MetricName: "nagios_test",
 	}
 
-	// invalidOutput3
-	metrics, err := parser.Parse([]byte(invalidOutput3))
-	require.NoError(t, err)
-	require.Len(t, metrics, 0)
+	tests := []struct {
+		name    string
+		input   string
+		inputF  func(string) []byte
+		assertF func(*testing.T, []telegraf.Metric, error)
+	}{
+		{
+			name: "valid output 1",
+			input: `PING OK - Packet loss = 0%, RTA = 0.30 ms|rta=0.298000ms;4000.000000;6000.000000;0.000000 pl=0%;80;90;0;100
+This is a long output
+with three lines
+`,
+			assertF: func(t *testing.T, metrics []telegraf.Metric, err error) {
+				require.NoError(t, err)
+				require.Len(t, metrics, 3)
+				// rta
+				assert.Equal(t, map[string]string{
+					"unit":     "ms",
+					"perfdata": "rta",
+				}, metrics[0].Tags())
+				assert.Equal(t, map[string]interface{}{
+					"value":       float64(0.298),
+					"warning_lt":  float64(0),
+					"warning_gt":  float64(4000),
+					"critical_lt": float64(0),
+					"critical_gt": float64(6000),
+					"min":         float64(0),
+				}, metrics[0].Fields())
+
+				// pl
+				assert.Equal(t, map[string]string{
+					"unit":     "%",
+					"perfdata": "pl",
+				}, metrics[1].Tags())
+				assert.Equal(t, map[string]interface{}{
+					"value":       float64(0),
+					"warning_lt":  float64(0),
+					"warning_gt":  float64(80),
+					"critical_lt": float64(0),
+					"critical_gt": float64(90),
+					"min":         float64(0),
+					"max":         float64(100),
+				}, metrics[1].Fields())
+
+				assertNagiosState(t, metrics[2], map[string]interface{}{
+					"state":   int64(0),
+					"msg":     "PING OK - Packet loss = 0%, RTA = 0.30 ms",
+					"longmsg": "This is a long output with three lines",
```

Comment:
Would it make sense to name these: `service_output` and `long_service_output` to match the `$SERVICEOUTPUT$` and `$LONGSERVICEOUTPUT$` macros they correspond to?

Secondly, do you think we should include the `\n` espaced newlines like nagios would?
```
This is a long output\nwith three lines
```

Or in the example from the nagios documentation:
```
/ 15272 MB (77%);\n/boot 68 MB (69%);\n/var/log 819 MB (84%);
```

Reply:
- I like the `service_output` and `long_service_output`.

- I used the `\n` at first, then I saw https://github.com/influxdata/docs.influxdata.com/pull/2192, which says the line protocol doesn't support the `\n` inside fields. 
Now, that I think about it, the `influxdb` output plugin's work to properly handle this. Since there are other outputs possible, I will use the `\n` again.

Reply:
Yeah, just to clarify those docs the `\n` characters can be in a field but have no special meaning.